### PR TITLE
Prevent from node sync with itself

### LINKED ISSUE #336: Node tries to sync with itself
I've noticed that after some time after starting 0.5.0. Node stucks on getting blocks from itself.
Possible solutions:

- do not allow to sync from itself
- query multiple nodes for new block and pick fastest reply, possibly drop other connections if new block and it's data has been acquired / each iteration ask different peer with correct height

```
[inf] 2016-12-03 10:18:30 | Broadhash consensus now 1 %
[inf] 2016-12-03 10:18:43 | Finished sync
[inf] 2016-12-03 10:18:54 | Starting sync
[WRN] 2016-12-03 10:18:54 | EVERSION:0.4.1 Removing peer 52.58.156.220:8000 GET /blocks/common?ids=13088404568012503670,6892891772145182357,8341960644728909029,1134575201568741266,14928740171326799900,2786663189856285424,13658550407518916215
[inf] 2016-12-03 10:19:00 | Broadhash consensus now 1 %
[inf] 2016-12-03 10:19:00 | Looking for common block with: PUBLIC_IP_OF_MY_NODE
[WRN] 2016-12-03 10:19:01 | EVERSION:0.3.2 Removing peer 111.207.165.188:8000 GET /list
[inf] 2016-12-03 10:19:03 | Found common block: 6817700788986749089 with: PUBLIC_IP_OF_MY_NODE
[inf] 2016-12-03 10:19:03 | Loading blocks from: PUBLIC_IP_OF_MY_NODE
[inf] 2016-12-03 10:19:03 | Broadhash consensus now 1 %
[inf] 2016-12-03 10:19:15 | Finished sync
[inf] 2016-12-03 10:19:25 | Starting sync
[inf] 2016-12-03 10:19:32 | Broadhash consensus now 1 %
[inf] 2016-12-03 10:19:32 | Looking for common block with: PUBLIC_IP_OF_MY_NODE
[inf] 2016-12-03 10:19:36 | Found common block: 6817700788986749089 with: PUBLIC_IP_OF_MY_NODE
[inf] 2016-12-03 10:19:36 | Loading blocks from: PUBLIC_IP_OF_MY_NODE
[inf] 2016-12-03 10:19:36 | Broadhash consensus now 1 %
```
Restarting node fix problems temporary, usually for not so long.

### PR DESCRIPTION
**Pull request by MaciejBaj**
Secure that list of peers won't be extended with node's public address and port.

closes #336

### CODE REVIEW

File: test/api/peer.js [hunk 1]
```diff
@@ -50,6 +56,20 @@ describe('GET /peer/list', function () {
 				done();
 			});
 	});
+
+	it('should not accept himself for peer', function (done) {
```
Comment (Isabello) [inline test/api/peer.js]: Can you change this to `itself`

Comment (MaciejBaj) [inline test/api/peer.js]: sure :)

**Comment by karmacoma** [comment]
@MaciejBaj Please test these changes within local testnet environment first.

**Comment by MaciejBaj** [comment]
thanks @4miners for good remarks.

- Server have more than one public IP
Current implementation considers also multiple public addresses of the host. 
- Are you sure that our public IP will be broadcasted to other nodes after those changes?
Yes, I am sure that following function has nothing common with broadcasting IP to other nodes. 
- Private addresses / private testnet
It will be still impossible to peer with private nodes. We should discuss if we want to allow for such option and eventually create the separate issue on it.

**Comment by 4miners** [comment]
Looks good, few more scenarios to test:
- Can node sync/talk with other node - same server, same public IP, different port?
- Can node sync/talk with other node - same server, different public IP, different port?
- Can node sync/talk with private peers that I put in `config.peers.list`? And what happend if I put there peers from above cases?

**Comment by MaciejBaj** [comment]
So it will be possible to peer with peers from the same server (same or different IP) and different port (I think it's ok). Syncing with the peer that broadcasts private IP will be still impossible.

**Pull request merged**